#!/usr/bin/env python

PACKAGE='user_intentions'
import roslib; roslib.load_manifest(PACKAGE)

import sys
from math import pi

from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()

# gen.add("p_g_g_factor", double_t, 0, "The transition probability p_g_g_factor", 100, 1, 1000)
#gen.add("vx_samples", int_t, 0, "The number of samples to use when exploring the x velocity space", 20, 1, 300)
#gen.add("holonomic_robot", bool_t, 0, "Set this to true if the robot being controlled can take y velocities and false otherwise", True)

gen.add("p_g_g_factor", double_t, 0, "The transition probability p_g_g_factor", 100.0, 1.0, 1000.0)
gen.add("p_c_gx_sigma", double_t, 0, "The standard deviation of the command model p_c_gx_sigma", 0.95, 0.1, 5.0)
gen.add("proba_threshold", double_t, 0, "The confidence necessary to send a Goal", 0.9, 0.1, 1.0)
gen.add("discrete_interface", bool_t, 0, "To enable/disable the discrete mode Set true when using the keyboard", False)

exit(gen.generate(PACKAGE, "user_intentions", "UserIntentions"))